build.sbt

```diff
@@ -24,123 +24,166 @@ mimaFailOnNoPrevious := false
 // don't publish the aggregate root project
 publish / skip := true
 
-def commonProject(id: String, artifact: String, path: String): Project = {
-  Project(id, file(path)).settings(
-    name := artifact,
+def commonSettings(artifact: String): Seq[Setting[_]] = Seq(
+  name := artifact,
 
-    mimaPreviousArtifacts := Set(organization.value %% artifact % "2.6.0"),
-
-    scalacOptions := Seq(
-      // "-Xfatal-warnings", // some methods in Scala 2.13 are deprecated, but I don't want to maintain to copies of source
-      "-deprecation:false",
-      "-feature",
-      "-Xlint",
-      "-Ywarn-dead-code",
-      "-encoding", "UTF-8"
-    ),
-
-    // show full stack traces in test failures
-    Test / testOptions += Tests.Argument(TestFrameworks.ScalaTest, "-oF"),
-
-    // disable compilation of ScalaDocs, since this always breaks on links and isn't as helpful as source
-    Compile / doc / sources := Seq.empty,
-
-    // disable publishing empty ScalaDocs
-    Compile / packageDoc / publishArtifact := false,
+  mimaPreviousArtifacts := (
+    if (scalaBinaryVersion.value == "3") Set.empty // remove once _3 is published
+    else Set(organization.value %% artifact % "2.6.0")
+    )
+)
 
-    // Don't publish the test artifacts, nobody should depend on these
-    Test / publishArtifact := false,
+scalacOptions := Seq(
+  // "-Xfatal-warnings", // some methods in Scala 2.13 are deprecated, but I don't want to maintain to copies of source
+  "-deprecation:false",
+  "-feature",
+  "-encoding", "UTF-8"
+) ++ (if (scalaBinaryVersion.value == "3") Seq.empty else Seq(
+  "-Xlint",
+  "-Ywarn-dead-code",
+))
+
+// show full stack traces in test failures
+Test / testOptions += Tests.Argument(TestFrameworks.ScalaTest, "-oF")
+
+// disable compilation of ScalaDocs, since this always breaks on links and isn't as helpful as source
+Compile / doc / sources := Seq.empty
+
+// disable publishing empty ScalaDocs
+Compile / packageDoc / publishArtifact := false
```

Comment:
BLKR: These used to be applied to every project via `commonSettings()`. Now they are only applied to the root project and not the sub-projects. You can either add these back to all projects with `ThisBuild / configKey :=` or, if you don't want to apply it to every project, you can revert `commonSettings()` (and maybe rename it appropriately) and only apply it to the projects that you want these settings for. I'm not exactly sure what your goal with this change is though. If it is to add the settings to all the projects without interfering with project matrix, then this is a blocker comment, because this change will remove these settings from the projects and only apply them to the aggregate root project.

Reply:
fixed